packages/trader/src/AppV2/Components/ServiceErrorSheet/service-error-description.tsx

```diff
@@ -5,39 +5,54 @@ import { Text } from '@deriv-com/quill-ui';
 type ServiceErrorProps = {
     is_insufficient_balance: boolean;
     is_authorization_required: boolean;
-    services_error: { message?: string };
+    is_account_verification_required: boolean;
+    is_mf_verification_pending_modal_visible: boolean;
+    services_error_message?: string;
 };
 
 const ServiceErrorDescription: React.FC<ServiceErrorProps> = ({
     is_insufficient_balance,
     is_authorization_required,
-    services_error,
+    is_account_verification_required,
+    is_mf_verification_pending_modal_visible,
+    services_error_message,
```

Comment:
If we’re only rendering content based on one condition at a time, I don't think we need to be passing them all as props. We can consider consolidating them and pass as a single prop that represents the specific error type. For example, we can probably create a utility function inside `ServiceErrorSheet` component to return the error type and pass the error type to `ServiceErrorDescription` component.

Reply:
You mean to do something like this: to pass error type -> inside of ServiceErrorDescription render content based on type (switch case or if?)

Reply:
Yup, something like this in `ServiceErrorSheet` component.

```
const getErrorType = () => {
        if (is_insufficient_balance) return ServicesErrorType.InsufficientBalance:
        if (is_authorization_required) return ServicesErrorType.AuthorizationRequired
        // Additional cases for other types
        return null;
};
.
.
<ServiceErrorDescription error_type = {getErrorType()}/>
```